Content.Server/Shipyard/Systems/ShipyardSystem.cs

```diff
@@ -0,0 +1,178 @@
+using Content.Server.Shipyard.Components;
+using Content.Shared.Mobs.Components;
+using Content.Shared.GameTicking;
+using Robust.Server.GameObjects;
+using Robust.Server.Maps;
+using Robust.Shared.Map;
+using Robust.Shared.Map.Components;
+using Content.Shared.CCVar;
+using Robust.Shared.Configuration;
+using Content.Server.Cargo.Components;
+
+namespace Content.Server.Shipyard.Systems
+{
+
+    public sealed partial class ShipyardSystem : SharedShipyardSystem
+    {
+        [Dependency] private readonly IConfigurationManager _configManager = default!;
+        [Dependency] private readonly IMapManager _mapManager = default!;
+        [Dependency] private readonly PricingSystem _pricing = default!;
+        [Dependency] private readonly ShuttleSystem _shuttle = default!;
+        [Dependency] private readonly StationSystem _station = default!;
+        [Dependency] private readonly MapLoaderSystem _map = default!;
+        [Dependency] private readonly ShipyardConsoleSystem _shipyardConsole = default!;
+
+        public MapId? ShipyardMap { get; private set; }
+        private float _shuttleIndex;
+        private const float ShuttleSpawnBuffer = 1f;
+        private ISawmill _sawmill = default!;
+        private bool _enabled;
+
+        public override void Initialize()
+        {
+            _enabled = _configManager.GetCVar(CCVars.Shipyard);
+            _configManager.OnValueChanged(CCVars.Shipyard, SetShipyardEnabled);
+            _sawmill = Logger.GetSawmill("shipyard");
+            _shipyardConsole.InitializeConsole();
+            SubscribeLocalEvent<ShipyardConsoleComponent, ComponentInit>(OnShipyardStartup);
+            SubscribeLocalEvent<RoundRestartCleanupEvent>(OnRoundRestart);
+        }
+
+        private void OnShipyardStartup(EntityUid uid, ShipyardConsoleComponent component, ComponentInit args)
+        {
+            if (!_enabled)
+                return;
+
+            SetupShipyard();
+        }
+
+        private void OnRoundRestart(RoundRestartCleanupEvent ev)
+        {
+            _configManager.UnsubValueChanged(CCVars.Shipyard, SetShipyardEnabled);
+            CleanupShipyard();
+        }
+
+        private void SetShipyardEnabled(bool value)
+        {
+            if (_enabled == value)
+                return;
+
+            _enabled = value;
+
+            if (value)
+            {
+                SetupShipyard();
+            }
+            else
+            {
+                CleanupShipyard();
+            }
+        }
+
+        /// <summary>
+        /// Adds a ship to the shipyard, calculates its price, and attempts to ftl-dock it to the given station
+        /// </summary>
+        /// <param name="stationUid">The ID of the station to dock the shuttle to</param>
+        /// <param name="shuttlePath">The path to the shuttle file to load. Must be a grid file!</param>
+        public void PurchaseShuttle(EntityUid? stationUid, string shuttlePath, out ShuttleComponent? vessel)
```

Comment:
This should be a TryPurchaseShuttle function with a success bool if it's falliable, instead of just returning null.

Reply:
Also you can mark it as NotNullWhen(true)

Reply:
the TryPurchaseShuttle is wrapping around this function on a case by case basis and checking for additional parameters depending on usage? this is a back end function for either calling directly through the console for an admin or for further usage such as nukie base or functions that do not check for bank account, etc.

Reply:
ok i just rewrote the method in system to be in a Try function instead and handled it there